DeepSpeech.ipynb

```diff
@@ -1021,80 +1177,158 @@
-    "                \n",
-    "        # Prepare tensor board logging\n",
-    "        merged = tf.merge_all_summaries()\n",
-    "        writer = tf.train.SummaryWriter('%s/%s' % (\"logs\", time.strftime(\"%Y%m%d-%H%M%S\")), session.graph)\n",
+    "        # Loop over the batches\n",
+    "        for batch in range(total_batch/len(available_devices)):\n",
+    "            # Compute the average loss for the last batch\n",
+    "            _, batch_avg_loss = session.run([apply_gradient_op, tower_loss])\n",
+    "\n",
+    "            # Add batch to total_accuracy\n",
+    "            total_accuracy += session.run(accuracy)\n",
+    "\n",
+    "            # Log all variable states in current step\n",
+    "            step = epoch * total_batch + batch * len(available_devices)\n",
+    "            summary_str = session.run(merged)\n",
+    "            writer.add_summary(summary_str, step)\n",
+    "            writer.flush()\n",
     "        \n",
-    "        # Loop over the data set for training_epochs epochs\n",
-    "        for epoch in range(training_iters):\n",
-    "            # Define total accuracy for the epoch\n",
-    "            total_accuracy = 0\n",
-    "            \n",
-    "            # Loop over the batches\n",
-    "            for batch in range(total_batch/len(available_devices)):\n",
-    "                # Compute the average loss for the last batch\n",
-    "                _, batch_avg_loss = session.run([apply_gradient_op, tower_loss])\n",
-    "                \n",
-    "                # Add batch to total_accuracy\n",
-    "                total_accuracy += session.run(accuracy)\n",
-    "                \n",
-    "                # Log all variable states in current step\n",
-    "                step = epoch * total_batch + batch * len(available_devices)\n",
-    "                summary_str = session.run(merged)\n",
-    "                writer.add_summary(summary_str, step)\n",
-    "                writer.flush()\n",
-    "\n",
-    "            # Print progress message\n",
-    "            if epoch % display_step == 0:\n",
-    "                print \"Epoch:\", '%04d' % (epoch+1), \"avg_cer=\", \"{:.9f}\".format((total_accuracy / total_batch))\n",
-    "            \n",
-    "            # Checkpoint the model\n",
-    "            if (epoch % checkpoint_step == 0) or (epoch == training_iters - 1):\n",
-    "                checkpoint_path = os.path.join(checkpoint_dir, 'model.ckpt')\n",
-    "                print \"Checkpointing in directory\", \"%s\" % checkpoint_dir\n",
-    "                saver.save(session, checkpoint_path, global_step=epoch)\n",
+    "        # Print progress message\n",
+    "        if epoch % display_step == 0:\n",
+    "            print \"Epoch:\", '%04d' % (epoch+1), \"avg_cer=\", \"{:.9f}\".format((total_accuracy / total_batch))\n",
+    "            _, last_train_wer = print_wer_report(session, \"Training\", tower_decodings, tower_labels)\n",
```

Comment:
Another good catch! However: I took a different approach by re-introducing a feed_dict. I think it turned out to be quite elegant. Please have a look.